src/WebJobs.Script/Description/DotNet/NoOpFunctionMetadataResolver.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Collections.ObjectModel;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Scripting;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    public sealed class NoOpFunctionMetadataResolver : IFunctionMetadataResolver
+    {
+        public ScriptOptions CreateScriptOptions()
+        {
+            return ScriptOptions.Default;
+        }
+
+        public IReadOnlyCollection<string> GetCompilationReferences()
+        {
+            return new ReadOnlyCollection<string>(new List<string>());
```

Comment:
Don't new these up each time. Definitely you don't need to new up the inner list. Can't you return something derived from Enumerable.Empty<string>?

Reply:
ahhh right right! Yeah that makes sense